engine/time/src/main/java/io/deephaven/time/DateTimeUtils.java

```diff
@@ -2567,14 +2569,214 @@ public static long nanosOfDay(@Nullable final Instant instant, @Nullable final Z
      * @param dateTime time
      * @return {@link QueryConstants#NULL_LONG} if either input is {@code null}; otherwise, number of nanoseconds that
      *         have elapsed since the top of the day
+     * @deprecated Use {@link #nanosOfDay(ZonedDateTime, boolean)} instead.  To be removed soon.
      */
     @ScriptApi
+    @Deprecated
     public static long nanosOfDay(@Nullable final ZonedDateTime dateTime) {
+        return nanosOfDay(dateTime, false);
+    }
+
+    /**
+     * Returns the number of milliseconds that have elapsed since the top of the day.
+     * <p>
+     * On days when daylight savings time events occur, results may be different from what is expected based upon the
+     * local time. For example, on daylight savings time change days, 9:30AM may be earlier or later in the day based
+     * upon if the daylight savings time adjustment is forwards or backwards.
+     *
+     * @param instant time
+     * @param timeZone time zone
+     * @return {@link QueryConstants#NULL_INT} if either input is {@code null}; otherwise, number of milliseconds that
+     *         have elapsed since the top of the day
+     * @deprecated Use {@link #millisOfDay(Instant, ZoneId, boolean)} instead.  To be removed soon.
+     */
+    @ScriptApi
+    @Deprecated
+    public static int millisOfDay(@Nullable final Instant instant, @Nullable final ZoneId timeZone) {
+        return millisOfDay(instant, timeZone, false);
```

Comment:
I don't know what the culture is here, but I'm feeling a little sad that we have to provide all these `${UNIT}OfDay` methods. Would it be ok just to provide `nanosOfDay` and let our user call `nanosTo${UNIT}` e.g. `nanosToSeconds` or whatever?

Just an observation, I'm not deeply opinionated on this.